### ISSUE TITLE
Duplicate discussions created when data-strict is enabled

### ISSUE THREAD

**Issue by andrewlock** [giscus/giscus#738]
Hi, I recently switched my blog to using giscus, so thank you so much for creating a great tool!

Unfortunately I've just noticed that giscus appears to be creating duplicate discussions for a post, even though I have `data-strict` set. I've also restricted to a single origin, so I know it's not a duplicate somewhere creating them either...

My configuration looks like this:

```html
<script src="https://giscus.app/client.js"
        data-repo="andrewlock/blog-comments"
        data-repo-id="R_kgDOHvS21g"
        data-category="Announcements"
        data-category-id="DIC_kwDOHvS21s4CQg24"
        data-mapping="pathname"
        data-strict="1"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-input-position="top"
        data-theme="preferred_color_scheme"
        data-lang="en"
        data-loading="lazy"
        crossorigin="anonymous"
        async>
</script>
```

This is an example post that has a problem: 
https://andrewlock.net/adding-validation-to-strongly-typed-configuration-objects-in-dotnet-6/

And these are the two discussions it created:
- https://github.com/andrewlock/blog-comments/discussions/47
- https://github.com/andrewlock/blog-comments/discussions/48

I don't suppose you have any idea what's going on? thanks very much for your help!

**Comment by laymonage** [comment]
Thanks for the report!

Looks like duplicate requests were made around the same time... Not sure how this could happen, as we disable the comment button and the reaction button if a discussion creation request is still being sent. I'll try to look into this later.

**Comment by andrewlock** [comment]
Thanks! I wonder, if two users loaded the page, and one sends the creation request, presumably the second user's comments section assumes it needs to create the discussion, and so creates another one. Is that a possibility? 🤔

**Comment by laymonage** [comment]
Indeed it is, but it doesn't seem to be what happened in your case. Both discussions were created at the same minute, but this comment is the only one with the same timestamp as the discussion creation: https://github.com/andrewlock/blog-comments/discussions/48#discussioncomment-3795551

The first comment on the other discussion is a few minutes later: https://github.com/andrewlock/blog-comments/discussions/47#discussioncomment-3833670

There are a couple of reactions though, so might be from there as well, but what are the chances of the reactions being submitted as the first comment on the other discussion?...

I also found another instance of this happening, though there are no comments:
- https://github.com/andrewlock/blog-comments/discussions/68
- https://github.com/andrewlock/blog-comments/discussions/69

I assume that was you, pregenerating the discussion by commenting then deleting the comment? Not sure how you did that. though.

I'm pretty sure this is reproducible by opening two tabs, leaving an initial comment on one tab, and another comment on the other tab after the first one has been submitted.

Maybe I should make it so that giscus' creation API searches for a matching discussion before actually sending a create mutation to GitHub...

**Comment by andrewlock** [comment]
> I also found another instance of this happening, though there are no comments:
> I assume that was you, pregenerating the discussion by commenting then deleting the comment? Not sure how you did that. though.

No, That wasn't me! I haven't tried doing anything like that... I guess someone else may have done? 🤔 

> Maybe I should make it so that giscus' creation API searches for a matching discussion before actually sending a create mutation to GitHub...

That seems like a good workaround to me! There would still be an _outside_ chance of duplicate comments if the requests overlap, but that seems a _very_ small chance, unlike now where it's easy to reproduce with it open in two tabs...

Even if there's something else weird going on here (evidenced by the discussions with no comments), that workaround may well resolve it 🙂 

Thanks for looking into it!

**Comment by yihui** [comment]
> Maybe I should make it so that giscus' creation API searches for a matching discussion before actually sending a create mutation to GitHub...

I also think that's definitely a great idea---the decision on whether to create a new discussion should be made *right before actually creating it*, instead of at the moment when giscus is loaded on the page. For a popular site, the latter can easily lead to duplicate discussions (many people reading the page at the same time but they could finish typing out the comments at any time).

Thanks for the great work!
